cpp/dbscan.cpp

```diff
@@ -68,9 +72,13 @@ auto Dbscan::fit_predict(std::vector<Dbscan::Point> const& points) -> std::vecto
     std::vector<std::uint32_t> new_point_to_point_index_map(std::size(points));
     std::uint32_t i{0};
     for (auto const& pt : points) {
-        auto const bin_x{static_cast<std::uint32_t>(std::floor((pt[0] - min[0]) / eps_))};
-        auto const bin_y{static_cast<std::uint32_t>(std::floor((pt[1] - min[1]) / eps_))};
+        auto bin_x{static_cast<std::uint32_t>(std::floor((pt[0] - min[0]) / eps_))};
+        auto bin_y{static_cast<std::uint32_t>(std::floor((pt[1] - min[1]) / eps_))};
+
+        bin_x = bin_x < num_bins_x ? bin_x : num_bins_x - 1;
```

Comment:
Just FYI, using `std::min` here would be more readable and allow you to keep the `const`.